src/storage/statistics_sr.c

```diff
@@ -229,26 +229,11 @@ xstats_update_statistics (THREAD_ENTRY * thread_p, OID * class_id_p, bool with_f
     }
   (void) catalog_end_access_with_dir_oid (thread_p, &catalog_access_info, NO_ERROR);
 
-  npages = estimated_nobjs = 0;
-
-  /* do not use estimated npages, get correct info */
-  if (file_get_num_user_pages (thread_p, &(cls_info_p->ci_hfid.vfid), &npages) != NO_ERROR)
-    {
-      goto error;
-    }
-  assert (npages > 0);
+  /* get npages and nobjs. do not use estimation, get correct info */
+  npages = nobjs = length = 0;
+  heap_get_num_objects (thread_p, &(cls_info_p->ci_hfid), &npages, &nobjs, &length);
```

Comment:
In the comment of heap_get_num_objects(), _This function is expensive since all pages of the heap are fetched to find the number of objects._

That's why this function got estimated numbers of objects. Could this change have a big performance impact?

Reply:
Existing estimates do not apply any deletion at all, which can result in very inaccurate statistics.